evm/src/cpu/kernel/asm/fields/fp12.asm

```diff
@@ -0,0 +1,103 @@
+global test_mul_Fp12:
+    // stack: f, f', g, g'
+    %store_fp6(0)
+    %store_fp6(6)
+    %store_fp6(12)
+    %store_fp6(18)
+    PUSH return_on_stack
+    // stack: return_on_stack
+    %jump(mul_Fp12)
+return_on_stack:
+    // stack:
+    %load_fp6(30)
+    %load_fp6(24)
+    // stack: h, h'
+    %jump(0xdeadbeef)
+
+
+///  macro | num | ops | cost
+///  -------------------------
+///  load  |   8 |  40 |  320
+///  store |   5 |  40 |  200
+///  dup   |   5 |   6 |   30
+///  swap  |   4 |  16 |   64
+///  add   |   3 |  16 |   48
+///  sub   |   2 |  17 |   34
+///  mul   |   3 | 156 |  468
+///  i9    |   1 |   9 |    9
+///  jump  |   1 |   1 |    1
+///
+/// TOTAL: 1174
+
+
+/// F = f + f'z
+/// G = g + g'z
+///
+/// H = h + h'z = FG 
+///
+/// h  = fg + sh(f'g')
+/// h' = (f+f')(g+g') - fg - f'g'
+///
+/// Note: each symbol in the stack comments consists of six words
+
+global mul_Fp12:
```

Comment:
It looks like this function assumes that the inputs are at addresses `0..12` and `12..24`, with the output written to `24..36`. The issue with this approach is that it makes the caller copy a whole bunch of data in order to call this function.

Usually when a function operates on data that has to be read from/written to memory, it would take pointers to the inputs and a pointer to the output buffer. So this function would take three pointers:
- first input
- second input
- the place where the output should be written